src/Streamly/Internal/Data/Stream/IsStream/Common.hs

```diff
@@ -433,7 +433,7 @@ takeWhile p m = fromStreamS $ S.takeWhile p $ toStreamS m
 
 {-# INLINE takeEndBy #-}
 takeEndBy :: (IsStream t, Monad m) => (a -> Bool) -> t m a -> t m a
-takeEndBy p m = fromStreamS $ S.takeEndBy p $ toStreamS m
+takeEndBy p m = fromStreamS $ D.takeEndBy p $ toStreamS m
```

Comment:
You need to use fromStreamD and toStreamD instead of fromStreamS et al. Streamk CI is failing because of this.